### PR TITLE
Restructure users

### PR DESCRIPTION
**Pull request by SimonLab**
# Ready

Merge the type owner and user in on type. see #419

### CODE REVIEW

**Comment by nelsonic** [comment]
@SimonLab let me know when this is ready for review. 👍

**Comment by SimonLab** [comment]
@nelsonic Do you have any idea why codecov complains? All the tests are passing but I don't understand yet what is exactly codecov/changes

**Comment by nelsonic** [comment]
@SimonLab re-run the tests on Travis.

**Comment by SimonLab** [comment]
@nelsonic I'm still stuck with codecov/changes. From what I understand codecov raise an alert if it saw an unexpected changes in coverage in files. From the detail of the failed codecov test, it seems that codecov is stuck on lines where I changed the name of an environment variable or when I require another function. Do you have any advises on this issue. This PR contain quiet a lot of changes and commits do you think it might be the reason of the problem. Also I see codcov/changes more like an indicator not limit to not merge, am I right?
Other than that the code is ready for review.

**Comment by SimonLab** [comment]
@nelsonic this PR is ready now, a bug on codecov was making the codecov/changes fail.

**Comment by iteles** [comment]
@SimonLab Now that we have a staging area, should we merge to there first, get Adam to test and then merge to master?

File: lib/dashboard.js [hunk 1]
```diff
@@ -15,7 +19,7 @@ exports.register = function (server, option, next) {
         if (!request.auth.isAuthenticated) {
           return reply.redirect('/login');
         } else {
-          var id = request.auth.credentials.id;
+          var id = (request.auth.credentials.id).toString();
```
Comment (nelsonic) [inline lib/dashboard.js]: you should not need the brackets around `request.auth.credentials.id` ...

File: lib/database-helpers/elasticsearch/multiget_jobs.js [hunk 2]
```diff
@@ -13,7 +13,14 @@ module.exports = function (ids, callback) {
         docs: params
       }
     }, function(error, response){
-      let result = response.docs.map(job => job._source);
+      let result = response.docs.map(job => {
```
Comment (nelsonic) [inline lib/database-helpers/elasticsearch/multiget_jobs.js]: `let` and `var` are identical when used in `function` scope. http://stackoverflow.com/questions/762011/let-keyword-vs-var-keyword

File: lib/handlers/clients/get_detailed_client.js [hunk 3]
```diff
@@ -13,14 +13,14 @@ module.exports = function (request, reply) {
...
     getClient(request.params.id, function (errClient, client) {
 
-      getOwners(function (errOwners, owners) {
+      listUsers(function (errOwners, users) {
 
-        var accountManagers = owners.filter(function(owner) {
+        var accountManagers = users.filter(function(user) {
 
-          return owner.id === client.accountManager;
+          return user.id.toString() === client.accountManager.toString();
```
Comment (nelsonic) [inline lib/handlers/clients/get_detailed_client.js]: are these values not stored as `String` by default?

File: lib/helpers/generate_signature.js [hunk 4]
```diff
@@ -30,12 +30,12 @@ var signature = "";
   signature += '</td>'
   signature += '<td width="274" valign="top" style="width:205.55pt;border:none;padding:0cm 5.4pt">'
   signature += '<p style="margin-bottom:0.0001pt">'
-  signature += '<span style="font-size:14pt"><font face="arial, helvetica, sans-serif"><font color="#999999">' + user.fn + '</font><br></font></span>'
+  signature += '<span style="font-size:14pt"><font face="arial, helvetica, sans-serif"><font color="#999999">' + user.names.fullname + '</font><br></font></span>'
```
Comment (nelsonic) [inline lib/helpers/generate_signature.js]: this should either be multi-lined with a [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) or put in a handlebars template.

File: test/users.test.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+require('env2')('.env');
```
Comment (nelsonic) [inline test/users.test.js]: can you simply require `env2` *once* in `../lib/index.js` ...?

File: test/z-google-oauth-handler.test.js [hunk 6]
```diff
@@ -91,7 +91,45 @@ describe('Mock /googleauth?code=oauth2codehere', function () {
...
+      });
+    });
+  });
+});
+
+describe('Mock /googleauth?code=oauth2codehere', function () {
+
+  it('Get a google token with a right code and the right permission with admin true', function (done) {
+
+    var token_fixture = fs.readFileSync(__dirname + '/fixtures/sample-auth-token.json');
```
Comment (nelsonic) [inline test/z-google-oauth-handler.test.js]: you don't need to `fs.readFileSync` when the file is `.json` you can simple `require()` it ... 😉

File: views/helpers/adminCheckbox.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+const Handlebars = require('handlebars');
+
+module.exports = function (isAdmin) {
+
+  let result = "<input type=\"checkbox\" name=\"admin\" > Admin <br/>";
+
+  if(isAdmin === true) {
+
+    result = "<input type=\"checkbox\" name=\"admin\" checked> Admin <br/>";
```
Comment (nelsonic) [inline views/helpers/adminCheckbox.js]: try:
```js
result = '<input type="checkbox" name="admin" checked> Admin <br/>'
```

**Comment by nelsonic** [comment]
@SimonLab & @heron2014 the code looks good. nice work. ✨ 
I've left a few comments, feel free to make the changes in the next PR. 👍

**Pull request merged**